app/src/main/java/com/amaze/filemanager/ui/activities/MainActivity.java

```diff
@@ -997,16 +1005,17 @@ public boolean onOptionsItemSelected(MenuItem item) {
 
     // Handle action buttons
     final MainFragment mainFragment = getCurrentMainFragment();
-    Objects.requireNonNull(mainFragment);
 
     switch (item.getItemId()) {
       case R.id.home:
-        mainFragment.home();
+        Objects.requireNonNull(mainFragment).home();
```

Comment:
As I see in #2432   `Objects.requireNonNull` doesn't really prevent NPE. Maybe add a manual check or use `Optional#isPresent`, and in case of null give some toast, saying `operation failed`,
Need to do this everywhere in this class.

Reply:
Sorry but `java.util.Optional` is supported on API >= 24. Will do classic null check then.